docs/design/deployment_design.md

```diff
@@ -0,0 +1,199 @@
+<!--
+SPDX-License-Identifier: MIT
+Copyright (c) 2020 The Authors.
+
+Authors: Hong Chang        <@Hong-Chang>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:The above copyright
+notice and this permission notice shall be included in all copies or
+substantial portions of the Software.THE SOFTWARE IS PROVIDED "AS IS",
+WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
+THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+-->
+
+This doc is to discuss targets and approaches to deploy Mizar to kubernetes (k8s) and Arktos.
+
+### Targets
+
+- Mizar deployment for both k8s and Arktos.
+- Mizar deployment in different environment including local dev environment (one VM), GC  P environment (multiple VMs), AWS environment (multiple VMs).
+- Mizar deployment should support factors such as multi-apiserver, multi-etcd, multi-tenant
+- Support deploy two version of Mizar: for developer and for end user
+Mizar setup script should support update Mizar to new version.
+
+### Requirements
+
+- Mizar deployment should be easy for user. Only need one operation or one YAML file.
+- After deployment, user should be able to easily sanity check deployment result.
+
+### Definitions
+
+- **Mizar Phase 1**: current Mizar components are in the same cluster of k8s.
+
+  ![Simple Endpoint](png/mizar-phase-1.png)
+
+- **Mizar Phase 2**: Mizar is targeting to 2-cluster model. In the model, Mizar is running at a k8s or Arktos cluster, and it provides network functions to another k8s or Arktos cluster through Mizar proxy.
```

Comment:
Change "In the model, Mizar..." to: "In this 2-cluster model, Mizar's management plane is running in a separate k8s cluster than Arkos compute cluster. An Arktos cluster will deploy the transit daemon and a Mizar proxy to communicate with the Mizar management plane."

Reply:
Updated.